### PR TITLE
Update the Documentation with the new directory structure.

### PR DESCRIPTION
**Pull request by farmaazon**
[ci no changelog needed]

### Pull Request Description

This PR removes the old, outdated summary of crates structure, and put instead an overview to the CONTRIBUTING.md in the app/gui (it seemed most relevant). Also, the docs for main parts of the IDE were extended.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Review by mwu-tow — APPROVED** [review]

File: app/gui/docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -117,6 +117,45 @@ so via `npm install prettier`. To use it manually via command line run
...
+    layer. The library is used by both the desktop application and the cloud
+    environment. For further documentation see the documentation of the crate
+    (at the top of the `src/lib.rs` file).
+  - `ide-desktop`: The desktop version of the Enso IDE. Implemented as an
+    electron application which spawns backend services, loads the WASM gui
+    library and run the main entry point.
+  - `engine`: (In the future: see the note at the section beginning). The
+    implementation of the language itself: CLI tools like compiler or
+    interpreter, as well as the services used as backend for the Enso IDE
+    (Language Server and Project Manager). \*\*
```
Comment (mwu-tow) [inline app/gui/docs/CONTRIBUTING.md]: Why the backslashed stars here?

**Review by mwu-tow — APPROVED** [review]

File: app/gui/docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -117,6 +117,45 @@ so via `npm install prettier`. To use it manually via command line run
...
+The subdirectories of interests are:
+
+- `app`: The actual products delivered in this repository:
+  - `gui`: A rust crate compiled to a WASM library with all the logic of the GUI
+    layer. The library is used by both the desktop application and the cloud
+    environment. For further documentation see the documentation of the crate
+    (at the top of the `src/lib.rs` file).
+  - `ide-desktop`: The desktop version of the Enso IDE. Implemented as an
+    electron application which spawns backend services, loads the WASM gui
+    library and run the main entry point.
```
Comment (vitvakatu) [inline app/gui/docs/CONTRIBUTING.md]:
```suggestion
    library and runs the main entry point.
```

File: app/gui/docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -117,6 +117,45 @@ so via `npm install prettier`. To use it manually via command line run
...
 are plugins for many IDEs available to do this for you.
 
+### Repository Structure Overview
+
+**Note**: Currently, the Enso repository is going through a process of
+refactoring, and it is not finished yet: the Engine files are still not in the
+`app/engine` where they ought to be, but in the root directory instead.
+
+The root directory contains `Cargo.toml` and `build.sbt` files, allowing to open
+all rust or scala code as a single project in your favourite IDE. There is also
```
Comment (vitvakatu) [inline app/gui/docs/CONTRIBUTING.md]:
```suggestion
all rust or scala code as a single project in your favorite IDE. There is also
```

File: app/gui/docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -117,6 +117,45 @@ so via `npm install prettier`. To use it manually via command line run
...
+  - `gui`: A rust crate compiled to a WASM library with all the logic of the GUI
+    layer. The library is used by both the desktop application and the cloud
+    environment. For further documentation see the documentation of the crate
+    (at the top of the `src/lib.rs` file).
+  - `ide-desktop`: The desktop version of the Enso IDE. Implemented as an
+    electron application which spawns backend services, loads the WASM gui
+    library and run the main entry point.
+  - `engine`: (In the future: see the note at the section beginning). The
+    implementation of the language itself: CLI tools like compiler or
+    interpreter, as well as the services used as backend for the Enso IDE
```
Comment (vitvakatu) [inline app/gui/docs/CONTRIBUTING.md]:
```suggestion
    interpreter, as well as the services used as a backend for the Enso IDE
```

File: app/gui/docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -117,6 +117,45 @@ so via `npm install prettier`. To use it manually via command line run
...
+    (Language Server and Project Manager). \*\*
+- `lib`: All libraries not being a main component of our application. They are
+  grouped by the language. The most prominent are:
+  - `rust/prelude`: A library containing most popular utilities and imports.
+    Should be imported in each rust module - see Contributing guidelines.
+  - `rust/ensogl`: EnsoGL Framework for creating efficient GUI applications in
+    WASM.
+  - `rust/frp`: The library allowing following the Functional Reactive
```
Comment (vitvakatu) [inline app/gui/docs/CONTRIBUTING.md]:
```suggestion
- `lib`: All libraries not being the main components of our application. They are
  grouped by language. The most prominent are:
  - `rust/prelude`: A library containing the most popular utilities and imports.
    Should be imported in each rust module - see Contributing guidelines.
  - `rust/ensogl`: EnsoGL Framework for creating efficient GUI applications in
    WASM.
  - `rust/frp`: The library allows following the Functional Reactive
```

**Review by vitvakatu — COMMENTED** [review]
Slight suggestions from Grammarly, otherwise LGTM

File: app/gui/src/lib.rs [hunk 6]
```diff
@@ -1,5 +1,32 @@
-//! Main library crate for IDE. It includes implementation of
-//! controllers, view logic and code that wraps them all together.
+//! The Enso IDE GUI.
+//!
+//! This rust crate is compiled to WASM library with all the logic of the Enso IDE GUI layer.
+//! See README of the repository for the presentation of Enso IDE and its features.
+//!
+//! ## Where Things Starts
```
Comment (akavel) [inline app/gui/src/lib.rs]: also a minor typo

```suggestion
//! ## Where Things Start
```

**Review by vitvakatu — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

**Pull request merged**